src/functions.php

```diff
@@ -578,7 +578,7 @@ function select_server_for_aggregate_write_stage(Manager $manager, array &$optio
 
     /* If there is either no read preference or a primary read preference, there
      * is no special server selection logic to apply. */
-    if ($readPreference === null || $readPreference->getMode() === ReadPreference::RP_PRIMARY) {
+    if ($readPreference === null || $readPreference->getMode() === ReadPreference::PRIMARY) {
```

Comment:
Thanks Psalm for the trigger. Getting the primary mode integer without using the constant could be done like this, but I feel this is not optimal.
```suggestion
    static $primaryMode;
    $primaryMode = $primaryMode ?? (new ReadPreference(ReadPreference::PRIMARY))->getMode();
    
    if ($readPreference === null || $readPreference->getMode() === $primaryMode) {
```

I notice this function is not covered by tests.